### ISSUE TITLE
resnext speed

### ISSUE THREAD

**Issue by Mctigger** [Cadene/pretrained-models.pytorch#8]
Hey,
I tried resnext-101(32x4d) for 160x160 crops and also the torchvision resnet-101. For me resnext needs nearly double the time to train compared to the resnet. (Only thing I changed is the final GlobalAveragePooling and classifier layer)
Is this expected behavior?
[Here](https://github.com/facebookresearch/ResNeXt#single-crop-224x224-validation-error-rate) it says the required FLOPs are the same, but obviously frameworks like Pytorch may be slower because of a more complicated network structure)

**Comment by Cadene** [comment]
What do you mean by "double the time to train compared to the resnet" ? Double time to do a forward + backward pass ?

Also, usually those resnet / resnext networks take crops of size 160x160. Why do you use 160x160 crops ?

But anyway, you are right: because of obvious implementation reasons, the processing time may be higher for resnext, even if the amount of FLOPs is the almost the same.

**Comment by Mctigger** [comment]
Yea, training for a whole epoch. So just like you said forward + backward. My use case is on 160x160 crops and I wanted to try resnext. Your repository makes this very easy :)

Good to know. I just wanted to check I have not missed something or that my implementation is buggy.

**Comment by Cadene** [comment]
Edit: Also, usually those resnet / resnext networks take crops of size **224x224**. Why do you use 160x160 crops ?

**Comment by Mctigger** [comment]
I understand. But I don't want to train imagenet. I am using [transfer learning](http://cs231n.github.io/transfer-learning/) on a 180x180 dataset with 160x160 crops.

**Comment by Cadene** [comment]
I am surprised that it works with 160x160 inputs. If I was you, I would try to do an upsampling from 180x180 to 224x224 (maybe a bit more if you want to add some data augmentation) :P 

Regards

**Comment by Mctigger** [comment]
Yea, I already though about upsampling, but came to the conclusion that most probably it makes more sense to use dilated convolutions instead, since then my input is not interpolated, but I still maintain high resolution feature maps.

**Comment by Cadene** [comment]
Wow that's cool, did you get better results using dilated convolutions ?

**Comment by Mctigger** [comment]
I did not try any of this yet. These are just my thoughts. I actually do not know about a state of the art paper for classification using dilated convolutions. But in semantic segmentation they are often used to achieve a greater field of view. Intuitively it seems like in general the field of view can be increased by either some pooling operation/strided convolution (which reduces resolution, that's why 224 input is better than 160) or by using dilated convolutions (resolutions stays the same, but computation increases). 
Since I assume that low-level filters are not a accuracy-bottleneck, but mid-/high-level features are it may make sense to use dilated convolutions there. 
So I maybe will try to use the normal architecture for 160 -> 80 -> 40 -> 20 and then remove the remaining two pooling operations and use dilated convolutions on 20x20.

But like I said, I didn't try any of this yet!

**Comment by Cadene** [comment]
Please, let me know :)

**Comment by hengck23** [comment]
you need cudnn v7.

https://developer.nvidia.com/cudnn

What’s New in cuDNN 7?
Grouped Convolutions for models such as ResNeXt and Xception and CTC (Connectionist Temporal Classification) loss layer for temporal classification

https://discuss.pytorch.org/t/does-pytorch-optimize-the-group-parameter-in-convs/3060

**Comment by tstandley** [comment]
In case anyone is interested, I have an implementation of Xception here:
[https://github.com/tstandley/Xception-PyTorch](url)
Pretrained weights were converted with some difficulty from Keras's implementation.

**Comment by Cadene** [comment]
@tstandley it would be awesome to add it here, great work!

**Comment by Cadene** [comment]
@tstandley Done! https://github.com/Cadene/pretrained-models.pytorch/commit/4417b9eace32f7b8b8ed335407451011852a8dc9
